### PR TITLE
PR changes

### PR DESCRIPTION
**Pull request by karthikeyan7010**

### CODE REVIEW

File: features/NewCustomerRegistration.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Feature: New customer Registration
+
+
+Background: customer Registration
+	Given the user navigates to Homepage
+    And clicks on login create a account button
```
Comment (Vigneshmuthusamy) [inline features/NewCustomerRegistration.feature:6]: @karthikeyan7010 From homepage is there a button called "login creat a account"?

File: features/NewCustomerRegistration.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    
+Scenario: Empty Customer Registration Authentication (Negative scenario)
+	Given the user should be navigated to customer account page
```
Comment (Vigneshmuthusamy) [inline features/NewCustomerRegistration.feature:9]: @karthikeyan7010 If you set the background right, we don't need Given here

File: features/NewCustomerRegistration.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    When the user clicks on create an account button
+    Then the user should see 5 error message "This is a required field"
+
+Scenario: Emailid and Password validation for Customer Registration Authentication (Negative scenario)
+    When I fill the login form with "<emailid>"
+    Then the user should see "Please enter a valid email address Please enter a valid email address Ex: johndoe@domain.com"
+    And User enters password with invalid format
```
Comment (Vigneshmuthusamy) [inline features/NewCustomerRegistration.feature:16]: @karthikeyan7010 Where is the samepl password?

File: features/NewCustomerRegistration.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+Background: customer Registration
+	Given the user navigates to Homepage
+    And clicks on login create a account button
+    
+Scenario: Empty Customer Registration Authentication (Negative scenario)
+	Given the user should be navigated to customer account page
+    When the user clicks on create an account button
+    Then the user should see 5 error message "This is a required field"
+
+Scenario: Emailid and Password validation for Customer Registration Authentication (Negative scenario)
```
Comment (Vigneshmuthusamy) [inline features/NewCustomerRegistration.feature:13]: @karthikeyan7010 We can split this into multiple scenarios 1 for email & 1 for password. Each scenario should have multiple test inputs

File: features/NewCustomerRegistration.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    And User enters password with invalid format
+    Then the user should see "Minimum of different classes of characters in password is 3. Classes of characters: Lower Case, Upper Case, Digits, Special Characters" message
+    Examples:
+            | emailid | 
+            | karthikeyan3071997gmail.com | 
+
+
+
+Scenario: valid Registration Authentication (Postive scenario)
+    When The user enters valid credentials in the required fields for customer registration
```
Comment (Vigneshmuthusamy) [inline features/NewCustomerRegistration.feature:25]: @karthikeyan7010 Start with given and set condition as the user entered valid credentials.... and when user clicks create account is the action here.

File: features/NewCustomerRegistration.feature [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    And click create an account button
+    Then the user should be redirected to My account landing page
```
Comment (Vigneshmuthusamy) [inline features/NewCustomerRegistration.feature:27]: @karthikeyan7010 Expected is new account should be created and redirected to my account

File: step-definitions/step.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
+const { Given, When,Then,defineStep } = require('@cucumber/cucumber')
```
Comment (Vigneshmuthusamy) [inline step-definitions/step.js:1]: @karthikeyan7010 There should be separate step file for each feature.

File: page-objects/NewCustomerRegistration-page.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+async formsubmit() {
+    await page.click('button:has-text("Create an Account")');
+    await page.waitForTimeout(1000);
+
+}
+
+async myaccpage() {
+    await page.waitForSelector('.page-title')
```
Comment (Vigneshmuthusamy) [inline page-objects/NewCustomerRegistration-page.js:73]: @karthikeyan7010 Expectation shouldn't be waitForSelector alone. You need to check if the expected output is coming in that page

**Review by Vigneshmuthusamy — CHANGES_REQUESTED** [review]

**Comment by karthikeyan7010** [comment]
I will complete the bug by today end of the day  and I share my PR

**Pull request closed without merging**